File: DTFlinkBatch/src/main/java/com/kevin/java/MapDemo.java
package com.kevin.java;

import org.apache.flink.api.common.functions.MapFunction;
import org.apache.flink.api.java.DataSet;
import org.apache.flink.api.java.ExecutionEnvironment;
import org.apache.flink.api.java.operators.DataSource;

import java.util.ArrayList;

/**
 * @author caonanqing
 * @version 1.0
 * @description     map：每来一个元素则处理一个元素
 * @createDate 2020/3/9
 */
public class MapDemo {

    public static void main(String[] args) throws Exception {

        // 获取运行环境
        ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
        ArrayList<String> list = new ArrayList<>();
        list.add("kevin java");
        list.add("coco python");
        list.add("Mr.cao scala");

        DataSource<String> text = env.fromCollection(list);

        // 每来一条数据处理一条数据
        DataSet<String> map = text.map(new MapFunction<String, String>() {
            @Override
            public String map(String value) throws Exception {
                // 1.获取数据库连接,每条数据获取一个连接
                // 2.处理数据
                // 3.关闭连接
                return value;
            }
        });

        // 输出数据
        map.print();

    }
}

File: dt-flink-datastream/src/main/scala/com/kevin/scala/transform/MapDemo.scala
package com.kevin.scala.transform

import org.apache.flink.api.common.functions.MapFunction
import org.apache.flink.streaming.api.scala.StreamExecutionEnvironment

/**
 * @author caonanqing
 * @version 1.0
 * @description     map ： 输入一个元素，输出一个元素
 *             常用于数据集内的数据清洗和转换
 * @createDate 2020/5/16
 */
object MapDemo {

  def main(args: Array[String]): Unit = {

    // 获取运行环境
    val env = StreamExecutionEnvironment.getExecutionEnvironment
    import org.apache.flink.api.scala._

    // 创建数据集
    val dataStream = env.fromElements(("a", 3), ("b", 5), ("c", 1), ("a", 9), ("c", 7), ("a", 8), ("d", 2))

    // 方式一，直接传入计算表达式，
    val stream1 = dataStream.map(t => (t._1, t._2 + 1))

    // 方式二：定义map函数逻辑，完成数据处理操作
    val stream2 = dataStream.map(new MapFunction[(String,Int),(String,Int)] {
      override def map(t: (String, Int)): (String, Int) = {
        return (t._1,t._2 + 1)
      }
    })

    // 打印输出
    stream1.print("stream1")
    stream2.print("stream2")

    // 执行程序
    env.execute("MapDemo")
  }

}
